### DISCUSSION TITLE
CefSharp in Safe Exam Browser with Zoom integration - Issues with automatic computer audio

### DISCUSSION THREAD

**Discussion by bgresens** [cefsharp/CefSharp#4652] [Q&A]
We are using Safe Exam Browser (SEB) for testing and access a Moodle page with Zoom integrated.  The zoom control is launched with a setting enabled "Automatically join audio by computer when joining a meeting".  When this Moodle page is viewed in Chrome or Edge an alert pops up asking if the browser can access the Mic and once you select "Always" the computer audio is automatically connected when a meeting starts.  However, when using CefSharp inside SEB we want no such prompt and for the computer audio to connect automatically anyway.  We are 90% there...

Using the command line arg "use-fake-ui-for-media-stream" we ensure audio and video are automatically allowed, but zoom still prompts for connecting computer audio, despite the Zoom setting.  This prompt has become an issue and we want computer audio to connect automatically, but have no idea how.  Research on the web and in SEB has proven fruitless.

Any thoughts would be appreciated.  Here is our full list of command line args passed to CefSharp:

```c#
cefSettings.AcceptLanguageList = CultureInfo.CurrentUICulture.Name;
cefSettings.CachePath = appConfig.BrowserCachePath;
cefSettings.CefCommandLineArgs.Add("touch-events", "enabled");
cefSettings.LogFile = appConfig.BrowserLogFilePath;
cefSettings.LogSeverity = error ? LogSeverity.Error : (warning ? LogSeverity.Warning : LogSeverity.Info);
cefSettings.PersistSessionCookies = !settings.DeleteCookiesOnStartup || !settings.DeleteCookiesOnShutdown;
cefSettings.UserAgent = InitializeUserAgent();
cefSettings.CefCommandLineArgs.Add("enable-media-stream");
cefSettings.CefCommandLineArgs.Add("enable-usermedia-screen-capturing");
cefSettings.CefCommandLineArgs.Add("use-fake-ui-for-media-stream");
cefSettings.CefCommandLineArgs.Add("disable-features", "WebRtcHideLocalIpsWithMdns");
```

**Comment by amaitland** [comment]
Your can try removing those command line args and switching to using [PermissionHandler](https://cefsharp.github.io/api/118.6.x/html/T_CefSharp_Handler_PermissionHandler.htm)

There's a very very basic example at https://github.com/cefsharp/CefSharp/blob/ed1763ee9510dca9e54ab766ea7cf305f71d2aa2/CefSharp.Example/Handlers/ExamplePermissionHandler.cs#L13

Make sure to assign the handler instance e.g.

```c#
chromiumWebBrowser.PermissionHandler  = new ExamplePermissionHandler ();
```

If your still having problems after switching to permissions handler then you'll need to move the conversation over to https://www.magpcss.org/ceforum/index.php
